src/test/UserServiceTest.java

```diff
@@ -0,0 +1,133 @@
+package test;
+
+
+import main.User;
+import main.UserService;
+import org.junit.Assert;
+import org.junit.Test;
+import org.mockito.Mockito;
+import java.util.Objects;
+
+import static org.mockito.Mockito.when;
+
+/**
+ * Test class for UserService.java
+ */
+public class UserServiceTest {
+    private Fixture fixture = new Fixture();
+
+    @Test
+    public void testAddUser_addsUser()
+    {
+        fixture.givenUserIsInitialized("Ahmed",10,0);
+        fixture.whenUserIsAddedSuccessfully();
+        fixture.thenAssertionResultsInEquality(1);
+    }
+    @Test(expected = IllegalArgumentException.class)
+    public void testAddUser_ageIsNegative_doesNotAddUser_throwsException()
+    {
+        try{
+            fixture.givenUserIsInitialized("Ahmed",-10,0);
+            fixture.whenUserIsNotAddedSuccessfully();
+        }catch (IllegalArgumentException e)
+        {
+            fixture.thenAssertionResultsInEquality(0);
+            fixture.thenThrowIllegalArgumentException();
+        }
```

Comment:
we dont need try and catch in test method, as we discussed code implementation can be hidden inside the fixture